src/order/filter/basic.lean

```diff
@@ -640,6 +640,7 @@ section
 -- this section needs to be before applicative, otherwise the wrong instance will be chosen
 protected def monad : monad filter := { map := @filter.map }
 
+-- `sanity_check` false positive: we want to unfold this `def`, so it shouldn't be a `lemma`
```

Comment:
```suggestion
-- TODO / FIXME `sanity_check` false positive: we want to unfold this `def`, so it shouldn't be a `lemma`
```

Reply:
I don't think that there is any way for `sanity_check` to guess that this should be a `def`. I think, a proper way to fix this is to introduce a tag `@[insane]` that will take a list of `sanity_check` rule names to ignore for this `def`. E.g., `@[insane def]` here or `@[insane unused_arg]` on `category_theory.single_obj`.

Reply:
I wrote the comment as a warning "don't try to fix this" for the next person who'll run `#sanity_check` on this file.

Reply:
Can you explain why you want/need to unfold this definition?

Reply:
@fpvandoorn I'll try to understand this in a few minutes. I tried to make it a `theorem`, then something failed. I thought that the reason was "they're trying to unfold the definition".

Reply:
I'm sorry. It seems that `simp` inside the proofs failed to unfold the definitions of `bind` and `pure`. I'll add them to `simp only` set, and remove the misleading comment.

Reply:
Awesome, looks like you got it!